winum.el

```diff
@@ -340,6 +340,16 @@ WINDOW: if specified, the window of which we want to know the number.
 
 ;; Internal functions ----------------------------------------------------------
 
+(defsubst winum--set-window-vector (window-vector)
+  "Use WINDOW-VECTOR according to the current `winum-scope'."
+  (if (eq winum-scope 'frame-local)
+      (puthash (selected-frame)
+               (cons window-vector
+                     (make-hash-table :size winum--window-count))
+               winum--frames-table)
+    (setq winum--window-vector window-vector)
+    (clrhash winum--numbers-table)))
```

Comment:
No particular reason. I just took it along because it was part of the else branch of the widow-vector setting if that I've extracted here.
If it's not always needed I suggest adding an `&optional clear-numbers-table?` parameter that's t when updating and nil when enlarging.